src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -4251,6 +4251,28 @@ public void Fill_InvalidStartIndexCount_ThrowsArgumentOutOfRangeException(int ar
             AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => Array.Fill(new string[arrayLength], "", startIndex, count));
         }
 
+        public class Bar
+        {
+        }
+
+        public class Foo
+        {
+        }
+
+        [Fact]
+        public static void Fill_Casting()
```

Comment:
Add similar tests coverage for the other Array.Fill overload. If I am reading the code correctly, it has a bug that you should be able to find by adding the tests.

Reply:
bug fixedup in 07dd088911a